ideascube/monitoring/forms.py

```diff
@@ -83,13 +84,14 @@ class InventorySpecimenForm(forms.ModelForm):
 
     def clean_specimen(self):
         barcode = clean_barcode(self.cleaned_data['specimen'])
+        specimen_query = Q(barcode=barcode)|Q(serial=barcode)
         try:
-            specimen = Specimen.objects.get(barcode=barcode)
+            specimen = Specimen.objects.get(specimen_query)
```

Comment:
What if a specimen has its serial set to the same value as another specimen's barcode?